src/shared/hooks/useDataApi.ts

```diff
@@ -86,18 +90,23 @@ const useDataApi = (url?: string): State => {
     // actual request
     fetchData(url, undefined, source.token).then(
       // handle ok
-      (response: AxiosResponse) =>
-        dispatch({ type: ActionType.SUCCESS, response, originalURL: url }),
+      (response: AxiosResponse) => {
+        if (didCancel) return;
+        dispatch({ type: ActionType.SUCCESS, response, originalURL: url });
+      },
       // catch error
       (error: AxiosError) => {
-        if (axios.isCancel(error)) return;
+        if (axios.isCancel(error) || didCancel) return;
         dispatch({ type: ActionType.ERROR, error });
       }
     );
 
     // handle unmounting of the hook
     // eslint-disable-next-line consistent-return
-    return () => source.cancel();
+    return () => {
+      source.cancel();
+      didCancel = true;
```

Comment:
So did you have any concrete issue with that?
Because I remember on the first version of that file I had that same logic implemented, but after a discussion with @its-mahdi we came to the conclusion that `axios.isCancel(error)` was enough to catch a case where the fetch would have been canceled.

Reply:
Yeah, I only put this in because I was getting some issues thrown up when running jest tests.

Reply:
All good then 👍